docs/require-by-string-aliases.md

```diff
@@ -0,0 +1,251 @@
+# Require by String with Aliases
+
+## Summary
+
+We need to add intuitive alias and paths functionality to facilitate the grouping together of related Luau files into libraries and allow for future package managers to be developed and integrated easily. 
+
+## Motivation
+
+The Roblox engine does not currently support require-by-string. One motivation for this RFC is to consolidate require syntax and functionality between the Roblox engine and the broader Luau language itself.
+
+Luau itself currently supports a basic require-by-string syntax that allows for requiring Luau modules by relative or absolute path. Unfortunately, the current implementation does not support aliases.
+
+## Design
+
+#### Aliases
+
+Aliases can be used to bind an absolute or relative path to a convenient, case-sensitive name that can be required directly.
+
+```json
+"aliases": {
+    "Roact": "C:/LuauModules/Roact-v1.4.2"
+}
+```
+
+Based on the alias map above, you would be able to require Roact directly:
+
+```lua
+local Roact = require("Roact")
```

Comment:
Why are we dropping the `@` prefix?

Reply:
Oh did not realize that section was needed in this part. Whoops, fixing it.

Reply:
Still need to fix the require strings here.